plugins/woocommerce/src/Blocks/TemplateOptions.php

```diff
@@ -0,0 +1,39 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Blocks;
+
+use Automattic\WooCommerce\Blocks\Options;
+
+/**
+ * TemplateOptions class.
+ *
+ * @internal
+ */
+class TemplateOptions {
+
+	/**
+	 * Initialization method.
+	 */
+	public function init() {
+		add_action( 'after_switch_theme', array( $this, 'check_should_use_blockified_product_grid_templates' ), 10, 2 );
+	}
+
+	/**
+	 * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+	 * option need to be updated accordingly.
+	 *
+	 * @return void
+	 */
+	public function check_should_use_blockified_product_grid_templates() {
```

Comment:
We are currently storing "yes" or "no" values here which means when we are doing comparisons there's a lot of conversions going on to check whether they are truthy or falsey. We can optimize this whilst maintaining these old values by refactoring below which will save new values in the database as `1` or `0` instead.

```php
	public function check_should_use_blockified_product_grid_templates() {
		$option_name = Options::WC_BLOCK_USE_BLOCKIFIED_PRODUCT_GRID_BLOCK_AS_TEMPLATE;
		// We previously stored "yes" or "no" values. This will convert them to true or false.
		$option_value          = wc_string_to_bool( get_option( $option_name ) );
		$should_use_blockified = wc_current_theme_is_fse_theme();

		// We don't need to do anything if the user switched to a classic theme or if the option is already set to true.
		if ( ! $should_use_blockified || $option_value ) {
			return;
		}

		update_option( $option_name, $should_use_blockified );
	}
```

Reply:
Good catches! I updated it in 8ae1080c6f1da1188010f5b9624da03ab9b85861 and f76d28d6680c03ad72df01740bf0163d51c59e6a. I did a small change to keep its behavior as it was, until now we weren't updating the option when switching between block themes, so I kept it that way.